core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -196,12 +198,23 @@ private ArtifactSignerProvider loadSigners(
   }
 
   final Collection<ArtifactSigner> loadAzureSigners() {
-    final AzureKeyVault keyVault =
-        new AzureKeyVault(
-            azureKeyVaultParameters.getClientlId(),
-            azureKeyVaultParameters.getClientSecret(),
-            azureKeyVaultParameters.getTenantId(),
-            azureKeyVaultParameters.getKeyVaultName());
+    final AzureKeyVault keyVault;
+    if (azureKeyVaultParameters.getAuthenticationMode() == AzureAuthenticationMode.CLIENT_SECRET) {
```

Comment:
wondering if this is the wrong place to be doing this conditional construction - i.e. IMHO this class just wants to say "get me a vault - here's some config" - this class shouldn't necessarily understand the config.